packages/scandipwa/src/component/MyAccountMyWishlist/MyAccountMyWishlist.container.js

```diff
@@ -107,15 +114,30 @@ export class MyAccountMyWishlistContainer extends PureComponent {
         };
     }
 
-    async addAllToCart() {
-        const { moveWishlistToCart } = this.props;
+    setIsQtyUpdateInProgress(status) {
+        this.setState({ isQtyUpdateInProgress: status });
+    }
+
+    addAllToCart() {
+        const { isQtyUpdateInProgress } = this.state;
 
         if (!isSignedIn()) {
-            await Promise.reject();
+            Promise.reject();
         }
 
         this.setState({ isLoading: true });
 
+        if (isQtyUpdateInProgress) {
+            // wait for qty update request to finish before moving all items to cart
+            setTimeout(() => this.addAllToCartAsync(), UPDATE_WISHLIST_FREQUENCY);
+        } else {
+            this.addAllToCartAsync();
+        }
```

Comment:
Wanna to clarify. It is when we updating qty then we actually setting timeout for 0.8 sec to start adding to cart? What if user will have bad internet and whislist request will take more time then we expect?  Is it handled somehow or this is not expected and won't happen?

Reply:
You are right - this logic is not that secure. I replaced it with a more secure and version. Now `isUpdateInProgress` value is checked every 100ms, until it is false. When it is false, items are moved to cart. Stop condition is 10 sec timeout. Even if this condition is reached, we still move wishlist to cart (as we can do it with previous quantities saved).